feat(engine): 2414: explain "Class name cannot be '{0}'" error (#18)
mattpocock/ts-error-translator: packages/engine/errors/2414.md

---
original: "Class name cannot be '{0}'"
excerpt: "You can't give a class the name of '{0}' because it's protected by TypeScript."
---

> Instead of naming your classes based on TypeScript types,
> 
> ```ts
> class {0} {
> }
> 
> You should name your classes to something which is not a typescript type. Checkout types here, [Everyday Types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html)

Reviewer: ```suggestion
Instead of naming your classes based on TypeScript types...

```ts
class {0} {
}

...you should name your classes to something which is not a TypeScript type. Check out all the types in the docs on [Everyday Types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html).
```

---
original: "Class name cannot be '{0}'"

> excerpt: "I was expecting a class name which doesn't resemble any typescript type"

Reviewer: ```suggestion
excerpt: "You can't give a class the name of '{0}' because it's protected by TypeScript."
```

---

> Instead of naming your classes based on typescript types,
> 
> ```ts
> class any{
> }
> ```
> 
> or 
> 
> ```ts
> class string{
> }
> ```
> 
> or
> 
> ```ts
> class number{
> }
> ```

Reviewer: ```suggestion
Instead of naming your classes based on TypeScript types,

```ts
class {0} {
}
```
```

> You should name your classes to something which is not a typescript type. Checkout types here, https://www.typescriptlang.org/docs/handbook/2/everyday-types.html

Author: Done
